Scripts/OnPrem/certificateChange.sh

```diff
@@ -0,0 +1,273 @@
+        if [ $? -ne 0 ]; then
+            echo "FATAL: Failed to access PFX with given password"
+            echo "FATAL: Check password or try running above command manually"
+            exit 1
+        fi
+    else 
+        echo "  No password needed"
+    fi
+    
+    echo ""
+}
+
+# Check that the pfx has a valid san attribue
+check_san_from_pfx() {
+
+    echo "Checking the SAN attribue"
+    echo "  Executing: openssl pkcs12 -in "$PFX_FILE" -passin pass:**** -nokeys -clcerts -out tmp.pem"
+    # Extract the certificate from the PFX file
+    openssl pkcs12 -in "$PFX_FILE" -passin pass:$PFX_PASSWORD -nokeys -clcerts -out /tmp/tmp.pem
+
+    if [ $? -ne 0 ]; then
+        echo "FATAL: Failed to create pem formated file"
+        exit 1
+    fi
+
+    # Extract the SAN information
+    echo "  Extracting the SAN attribue: openssl x509 -in /tmp/tmp.pem -noout -text | grep -A 1 "Subject Alternative Name:" | tail -n1"
+    local san=$(openssl x509 -in /tmp/tmp.pem -noout -text | grep -A 1 "Subject Alternative Name:" | tail -n1)
+
+    if [ $? -ne 0 ]; then
+        echo "FATAL: Failed to read SAN info"
+        echo "Cleaning up /tmp/tmp.pem"
+        rm /tmp/tmp.pem
+        exit 1
+    fi
+
+    # Clean up the temporary certificate
+    echo "  Removing temp cert: /tmp/tmp.pem"
+    rm -f /tmp/tmp.pem
+
+    
+    # Print the SAN (or you can process it further if needed)
+    if [[ $san =~ $FQDN ]]; then
+        echo "  The certificate has a valid san attribute for $FQDN"
+    else
+        echo "FATAL: The certificate does not have a valid san attribute for: $FQDN"
+        echo "FATAL: The san attributes are: $san"
+        exit 1
+    fi
+
+    echo ""
+ 
+}
+
+
+# Convert the PFX to a PFX with no password
+function convert_pfx() {
+    
+    echo "Converting the PFX for installation"
+
+    # Run this inside the container so we do not have FIPS issues
+    echo "  Extracting cert in pem format."
+    echo "  Executing: openssl pkcs12 -in $PFX_FILE -out $INSIGHTS_DIR/tmp.pem -nodes -passin pass:***"
+    openssl pkcs12 -in $PFX_FILE -out $INSIGHTS_DIR/tmp.pem -nodes -passin pass:$PFX_PASSWORD
+
+    if [ $? -ne 0 ]; then
+        echo "FATAL: Extration command failed!"
+        echo "FATAL: Checking permissons on $PWD"
+        exit 1
+    fi
+
+    echo "  Executing: sudo chmod 744 $INSIGHTS_DIR/tmp.pem"
+        if [ $? -ne 0 ]; then
+        echo "FATAL: Failed to set perms on $INSIGHTS_DIR/tmp.pem"
+        echo "FATAL: Checking permissons on $PWD"
+        exit 1
+    fi
+
+    echo "  Creating new cert.pfx at $INSIGHTS_DIR/cert.pfx"
+    echo "    Checking if $INSIGHTS_DIR/cert.pfx exists"
+    if [[ -e "$INSIGHTS_DIR/cert.pfx" ]]; then
+        randomN=$RANDOM
+        echo "    The file $INSIGHTS_DIR/cert.pfx exists moving to $INSIGHTS_DIR/cert.pfx_$randomN"
+        echo "    Executing: sudo mv $INSIGHTS_DIR/cert.pfx $INSIGHTS_DIR/cert.pfx_$randomN"
+        sudo mv $INSIGHTS_DIR/cert.pfx $INSIGHTS_DIR/cert.pfx_$randomN
```

Comment:
should we use sudo_exec (add dzdo option)

Reply:
I took out sudo. We have this in the install script because of the auto install. I thought about it and its better to not code in a script if we can.